eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/list/ThrowingFixedSizeListFactory.java

```diff
@@ -0,0 +1,137 @@
+/*
```

Comment:
I wonder if we should hide these classes. Inner classes somewhere? Maybe even a dynamic proxy with a trivial invocation handler (always throws)?

Reply:
```java
public class ThrowingInvocationHandler  implements InvocationHandler{
    private String error;
    public  MyInvocationHandler(String error) {
        this.error = error;
    }
    public Object invoke(Object proxy, Method m, Object[] args) throws Throwable {
        throw new IllegalStateException(this.error);
    }
}
```
creating one looks like this:
```java
(FixedSizeListFactory) Proxy.newProxyInstance(Lists.class.getClassLoader(), 
  new Class[] { FixedSizeListFactory.class }, 
  new ThrowingInvocationHandler(error));
```

Advantages: shorter code, never have to implement any new methods and totally outside the public API.